spec/patterns/calculation_spec.rb

```diff
@@ -0,0 +1,147 @@
+RSpec.describe Patterns::Calculation do
+  require 'active_support/all'
+  require 'timecop'
+  after { Object.send(:remove_const, :CustomCalculation) if defined?(CustomCalculation) }
+
+  describe ".result" do
+    it "returns a result of calculation within #result method" do
+      CustomCalculation = Class.new(Patterns::Calculation) do
+        private
+
+        def result
+          50
+        end
+      end
+
+      expect(CustomCalculation.result).to eq 50
+    end
+
+    it "#result, #result_for and #calculate are aliases" do
+      CustomCalculation = Class.new(Patterns::Calculation)
+
+      expect(CustomCalculation.method(:result)).to eq CustomCalculation.method(:result_for)
+      expect(CustomCalculation.method(:result)).to eq CustomCalculation.method(:calculate)
+    end
+
+
+    it "exposes first argument as a subject" do
+      CustomCalculation = Class.new(Patterns::Calculation) do
+        private
+
+        def result
+          subject
+        end
+      end
+
+      expect(CustomCalculation.result('test')).to eq 'test'
+    end
+
+    it "exposes all keyword arguments using #options by default" do
+      CustomCalculation = Class.new(Patterns::Calculation) do
+        private
+
+        def result
+          [options[:arg_1], options[:arg_2]]
+        end
+      end
+
+      expect(CustomCalculation.result(nil, arg_1: 20, arg_2: 30)).to eq([20, 30])
+    end
+
+    it "caches result for 'set_cache_expiry_every' time" do
+      CustomCalculation = Class.new(Patterns::Calculation) do
+        set_cache_expiry_every 1.hour
+        @counter = 0
+
+        class << self
+          attr_accessor :counter
+        end
+
+        def result
+          self.class.counter += 1
+        end
+      end
+
+      expect(CustomCalculation.result).to eq 1
+      expect(CustomCalculation.result).to eq 1
+      Timecop.travel 2.hours.from_now do
+        expect(CustomCalculation.result).to eq 2
+        expect(CustomCalculation.result).to eq 2
+      end
+    end
+
+    it "caches result for every option passed" do
+      CustomCalculation = Class.new(Patterns::Calculation) do
+        set_cache_expiry_every 1.hour
+        @counter = 0
+
+        class << self
+          attr_accessor :counter
+        end
+
+        def result
+          self.class.counter += 1
+        end
+      end
+
+      expect(CustomCalculation.result(1)).to eq 1
```

Comment:
try not to use argument same as result, as it suggests correlation, when there is none